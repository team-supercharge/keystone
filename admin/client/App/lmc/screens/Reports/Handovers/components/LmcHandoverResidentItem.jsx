import React from 'react'
import PropTypes from 'prop-types'
import LmcTimelineRow from '../../../../components/LmcTimelineRow.jsx'

const LmcHandoverResidentItem = ({ data }) => {
    const { logs, resident } = data
    const chosenName = resident.preferredName || resident.name.first
    
    return (
        <div>
            <div style={styles.residentContainer}>
                <div 
                    className="lmc-profile-picture__handover" 
                    style={{
                        background: `url(${resident.picture || PLACEHOLDER_IMAGE})` 
                    }} 
                />
                <span style={styles.residentName}>
                    <span style={styles.preferredName}>{chosenName}</span> ({ `${resident.name.first} ${resident.name.last}` })
                </span>
            </div>
            <ul style={styles.logContainer}>
                { logs.map((log, i) => {
                    return (
                        <LmcTimelineRow
                            key={log.id}
                            log={log}
                            index={i}
                            total={logs.length}
                        /> 
                    )
                })}
            </ul>
        </div>
        
    )
}

const PLACEHOLDER_IMAGE = 'https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png';

const styles = {
    logContainer: {
		listStyleType: 'none',
        paddingLeft: 0,
        paddingBottom: 0,
        position: 'relative',
        top: 20,
        left: 50
    },
    residentContainer: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'flex-start'
    },
    residentName: {
        fontSize: 16,
        position: 'relative',
        top: 12,
        left: 20
    },
    preferredName: {
        fontWeight: 600
    }
}
LmcHandoverResidentItem.propTypes = {
    data: PropTypes.object.isRequired
}

export default LmcHandoverResidentItem