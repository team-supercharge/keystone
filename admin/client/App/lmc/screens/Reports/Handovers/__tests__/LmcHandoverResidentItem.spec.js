import React from 'react'
import { shallow } from 'enzyme'
import LmcHandoverResidentItem from '../components/LmcHandoverResidentItem.jsx'

describe('LmcHandoverResidentItem', () => {
    let wrapper
    let data

    beforeEach(() => {
        data = { 
            logs: [{ id: 'TestId', description: 'This is a log', createdBy: 'TestCarer' }],
            resident: { id: 'TestId2', preferredName: 'Testy', name: { first: 'Test' , last: 'Name' }, picture: 'TestPicture' }
        }

        wrapper = shallow(
            <LmcHandoverResidentItem
                data={data}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders the resident name and preferred name', () => {
        const nameText = 'Testy (Test Name)'
        expect(wrapper.text().includes(nameText)).toBe(true)
    })

    it('renders an LmcTimelineRow with the correct log', () => {
        const row = wrapper.find('LmcTimelineRow')
        expect(row.props().log).toEqual(data.logs[0])
    })
})