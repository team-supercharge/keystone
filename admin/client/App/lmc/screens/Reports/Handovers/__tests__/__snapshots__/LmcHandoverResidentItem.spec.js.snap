// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcHandoverResidentItem renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <div
        className="lmc-profile-picture__handover"
        style={
          Object {
            "background": "url(TestPicture)",
          }
        }
      />
      <span
        style={
          Object {
            "fontSize": 16,
            "left": 20,
            "position": "relative",
            "top": 12,
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": 600,
            }
          }
        >
          Testy
        </span>
         (
        Test Name
        )
      </span>
    </div>
    <ul
      style={
        Object {
          "left": 70,
          "listStyleType": "none",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "position": "relative",
          "top": 20,
        }
      }
    >
      <LmcTimelineRow
        index={0}
        log={
          Object {
            "createdBy": "TestCarer",
            "description": "This is a log",
            "id": "TestId",
          }
        }
        total={1}
      />
    </ul>
  </div>,
  "nodes": Array [
    <div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <div
          className="lmc-profile-picture__handover"
          style={
            Object {
              "background": "url(TestPicture)",
            }
          }
        />
        <span
          style={
            Object {
              "fontSize": 16,
              "left": 20,
              "position": "relative",
              "top": 12,
            }
          }
        >
          <span
            style={
              Object {
                "fontWeight": 600,
              }
            }
          >
            Testy
          </span>
           (
          Test Name
          )
        </span>
      </div>
      <ul
        style={
          Object {
            "left": 70,
            "listStyleType": "none",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "position": "relative",
            "top": 20,
          }
        }
      >
        <LmcTimelineRow
          index={0}
          log={
            Object {
              "createdBy": "TestCarer",
              "description": "This is a log",
              "id": "TestId",
            }
          }
          total={1}
        />
      </ul>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LmcHandoverResidentItem
        data={
          Object {
            "logs": Array [
              Object {
                "createdBy": "TestCarer",
                "description": "This is a log",
                "id": "TestId",
              },
            ],
            "resident": Object {
              "id": "TestId2",
              "name": Object {
                "first": "Test",
                "last": "Name",
              },
              "picture": "TestPicture",
              "preferredName": "Testy",
            },
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "logs": Array [
              Object {
                "createdBy": "TestCarer",
                "description": "This is a log",
                "id": "TestId",
              },
            ],
            "resident": Object {
              "id": "TestId2",
              "name": Object {
                "first": "Test",
                "last": "Name",
              },
              "picture": "TestPicture",
              "preferredName": "Testy",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover"
              style={
                Object {
                  "background": "url(TestPicture)",
                }
              }
            />
            <span
              style={
                Object {
                  "fontSize": 16,
                  "left": 20,
                  "position": "relative",
                  "top": 12,
                }
              }
            >
              <span
                style={
                  Object {
                    "fontWeight": 600,
                  }
                }
              >
                Testy
              </span>
               (
              Test Name
              )
            </span>
          </div>
          <ul
            style={
              Object {
                "left": 70,
                "listStyleType": "none",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "position": "relative",
                "top": 20,
              }
            }
          >
            <LmcTimelineRow
              index={0}
              log={
                Object {
                  "createdBy": "TestCarer",
                  "description": "This is a log",
                  "id": "TestId",
                }
              }
              total={1}
            />
          </ul>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover"
              style={
                Object {
                  "background": "url(TestPicture)",
                }
              }
            />
            <span
              style={
                Object {
                  "fontSize": 16,
                  "left": 20,
                  "position": "relative",
                  "top": 12,
                }
              }
            >
              <span
                style={
                  Object {
                    "fontWeight": 600,
                  }
                }
              >
                Testy
              </span>
               (
              Test Name
              )
            </span>
          </div>
          <ul
            style={
              Object {
                "left": 70,
                "listStyleType": "none",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "position": "relative",
                "top": 20,
              }
            }
          >
            <LmcTimelineRow
              index={0}
              log={
                Object {
                  "createdBy": "TestCarer",
                  "description": "This is a log",
                  "id": "TestId",
                }
              }
              total={1}
            />
          </ul>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcHandoverResidentItem
    data={
      Object {
        "logs": Array [
          Object {
            "createdBy": "TestCarer",
            "description": "This is a log",
            "id": "TestId",
          },
        ],
        "resident": Object {
          "id": "TestId2",
          "name": Object {
            "first": "Test",
            "last": "Name",
          },
          "picture": "TestPicture",
          "preferredName": "Testy",
        },
      }
    }
  />,
}
`;
